R/github.R

```diff
@@ -279,3 +285,21 @@ check_gh_token <- function(auth_token) {
   }
   auth_token
 }
+
+detect_protocol <- function(credentials) {
+  # If a non-null credentials just return ssh
+  if (!is.null(credentials)) {
+    return("ssh")
```

Comment:
It is possible to pass credentials to git2r other than ssh. There's a whole family of functions: `cred_env()`, `cred_ssh_key()`, `cred_token()`, and `cred_user_pass()`. I do use them when git2r is being particularly obtuse about finding credentials. I wonder if this is a bit too simple? I.e. will still end up with lots of https people being treated as ssh people.

Reply:
We could cache in global git settings or strongly encourage people to set their `usethis.protocol` option. Maybe we should double down on the latter in this case, since we already support it.